shared/testutil/block.go

```diff
@@ -0,0 +1,369 @@
+package testutil
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/prysmaticlabs/go-bitfield"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// GenerateFullBlock generates a fully valid block with the requested parameters.
+// change the BeaconConfig() MAX_OPERATIONNAME lengths to build a block of the conditions
+// needed.
+func GenerateFullBlock(
```

Comment:
It might be easier to specify a config as an argument and use that instead of overriding the beacon config,  such as:

```go
func GenerateFullBlock(
    t *testing.TB
    bState *pb.BeaconState,
    privs []*bls.SecretKey,
    conf *BlockGenConfig,
)
```

where 
```go
type BlockGenConfig struct {
     MaxAttestations uint64
     ...
}

func DefaultBlockGenConfig() *BlockGenConfig {
    return &BlockGenConfig{
        MaxAttestations: params.BeaconConfig().MaxAttestations,
        ...
    }
}
```